src/app/feed/feed-post-icon-row/feed-post-icon-row.component.html

```diff
@@ -1,152 +1,350 @@
-        </a>
-      </ng-template>
-    </div>
-    <span *ngIf="!hideNumbers">{{ postContent.RepostCount + postContent.QuoteRepostCount }}</span>
-  </div>
-
-  <div
-    (click)="toggleLike($event)"
-    [ngClass]="{ 'fc-red': postContent.PostEntryReaderState ? postContent.PostEntryReaderState.LikedByReader : false }"
-    class="cursor-pointer d-flex align-items-center"
-  >
-    <i-feather
-      [ngClass]="{
-        is_animating: animateLike,
-        'heart-icon': postContent.PostEntryReaderState ? !postContent.PostEntryReaderState.LikedByReader : true,
-        'heart-icon-fill': postContent.PostEntryReaderState ? postContent.PostEntryReaderState.LikedByReader : false
-      }"
-      name="heart"
-      class="feed-post-icon-row__icon"
-    ></i-feather>
-    <span *ngIf="!hideNumbers">{{ postContent.LikeCount }}</span>
-  </div>
-  <ng-template #popTemplate>
-    <div id="diamond-popover" class="m-10px">
-      <div *ngIf="!sendingDiamonds">
-        <div class="mb-4" style="font-size: 11.25px">
-          {{ "feed_post_icon_row.give_diamond" | transloco }}
-          <b>@{{ postContent.ProfileEntryResponse.Username }}</b>
-          {{ "feed_post_icon_row.will_receive" | transloco }}
-        </div>
       </div>
+      <span *ngIf="!hideNumbers">{{ postContent.RepostCount + postContent.QuoteRepostCount }}</span>
     </div>
-  </ng-template>
-  <div class="cursor-pointer d-flex align-items-center">
-    <!--Container to hold the bounds for the mobile drag interface-->
+
     <div
-      class="diamond-mobile-drag-container unselectable"
+      *ngIf="!globalVars.isMobile(); else reactionSwiper"
+      (click)="sendReaction(allowedReactions[0], $event)"
       [ngClass]="{
-        'hide-diamonds':
-          !globalVars.loggedInUser?.PublicKeyBase58Check ||
-          postContent.ProfileEntryResponse?.PublicKeyBase58Check === globalVars.loggedInUser?.PublicKeyBase58Check
+        'fc-blue': myReactions.length > 0
       }"
+      class="position-relative cursor-pointer d-flex align-items-center pt-5px"
+      (mouseover)="debouncedToggleSelectReactionFunction(true)"
+      (mouseleave)="debouncedToggleSelectReactionFunction(false)"
     >
-      <div
-        [ngClass]="{ 'bg-danger': diamondDragCancel, show: diamondDragging }"
-        class="diamond-mobile-drag-instructions"
+      <div *ngIf="choosingReaction">
+        <ng-container *ngTemplateOutlet="reactionsPopTemplate"></ng-container>
+      </div>
+
+      <i-feather
+        [ngClass]="{
+          is_animating_ice: processedReaction,
+          'thumbs-up': postContent.PostEntryReaderState ? !postContent.PostEntryReaderState.LikedByReader : true,
+          'thumbs-up-fill': myReactions.length > 0
+        }"
+        name="thumbs-up"
+        class="feed-post-icon-row__icon thumbs-up"
+      ></i-feather>
+      <span *ngIf="!hideNumbers">{{ postReactionCounts.Total }}</span>
+    </div>
+
+    <ng-template #reactionSwiper>
+      <swipe-to-choose
```

Comment:
This is the component I built fully from the code that is used to send diamonds.
I think in the nearest future, we can also reuse this component for Diamond selection to reduce the code duplication. I didn't do it now, because was a bit afraid of breaking the existing functionality.

What do you think about it?